teps/0127-larger-results-via-sidecar-logs.md

```diff
@@ -0,0 +1,246 @@
+functionality identical and does not require any external storage mechanism.
+
+For further details, see the [demonstration][demo] of the [implementation][poc].
+
+This provides an opportunity to experiment with this solution to provide `Results` within the CRDs as we continue to
+explore other alternatives, including those that involve external storage.
+
+### Feature Gate
+
+This feature will be gated using a `result-extraction-method` feature flag. 
+
+Users have to set `result-extraction-method` to `"sidecar-logs"` to enable the feature:
+```shell
+kubectl patch cm feature-flags -n tekton-pipelines -p '{"data":{"result-extraction-method":"sidecar-logs"}}'
+```
+
+This feature is disabled by default. When disabled, `Results` continue to pass through `Termination Message`.
+
+### Logs Access
+
+This feature requires that the `Pipeline` controller has access to `Pod` logs. 
+
+Users have to grant `get` access to all `pods/log` to the `Pipeline` controller:
+```shell
+kubectl apply -f config/enable-log-access-to-controller/
+```
+
+### Size Limit
+
+The size limit per `Result` can be configured using the `max-result-size` feature flag, which takes the integer value
+of the bytes.
+
+The `max-result-size` feature flag defaults to 4096 bytes. This ensures that we support existing `Tasks` with only one
+`Result` that uses up the whole size limit of the `Termination Message`.
+
+If users want to set the size limit per `Result` to be something other than 4096 bytes, they can set `max-result-size`
+by setting `max-result-size: <VALUE-IN-BYTES>`. The value set here cannot exceed the CRD size limit of 1.5MB.
+
+```shell
+kubectl patch cm feature-flags -n tekton-pipelines -p '{"data":{"max-result-size":"<VALUE-IN-BYTES>"}}'
+```
+
+Even though the size limit per `Result` is configurable, the size of `Results` are limited by CRD size limit of 1.5MB.
+If the size of `Results` exceeds this limit, then the `TaskRun` will fail with a message indicating the size limit has
+been exceeded.
+
+### PipelineRun Status
+
+In [TEP-0100][tep-0100], we proposed changes to `PipelineRun` status to reduce the amount of information stored about
+the status of `TaskRuns` and `Runs`. As a result, the `PipelineRun` status is set up to handle larger `Results` without
+exacerbating the storage issues that were there before.
+
+For `ChildReferences` to be populated, the `embedded-status` must be set to `"minimal"`. We recommend that the minimal
+embedded status - `ChildReferences` - is enabled while migration is ongoing until it becomes the only supported status.
+This will ensure that larger `Results` from all its `TaskRuns` will not bloat the `PipelineRun` CRD.
+
+## Design Details
+
+The `Sidecar` will run a binary that:
+- receives argument for `Results`' paths and names which are identified from `taskSpec.results` field. This allows the
+  `Sidecar` to know the `Results` it needs to read.
+- has `/tekton/run` volume mounted where status of each `Step` is written.
```

Comment:
nit: we probably want to make sure the volumes are mounted as read only (https://github.com/tektoncd/pipeline/issues/4227)

Reply:
yes they are :). We should probably mention that.

Reply:
mentioned that it's read only 👍🏾